### ISSUE TITLE
Sliding Blurs or Animating Blurs Freezes Olive and Entire PC

### ISSUE THREAD

**Issue by DaniSeeh** [olive-editor/olive#376]
Whenever I use the slider to determine the radius of any of the blur effects, it completely freezes Olive for sometimes as long as several minutes, has crashed it once, and often freezes my entire desktop environment other than the mouse.

This also happens when I animate the blur with keyframes, though to a lesser extent.

Specs of machine:
Linux Mint 19.1
Ryzen 1700x
32GB ram
GTX 1070

**Comment by BrimsonBhin** [comment]
Yeah this is a common issue that was addressed in #205. It happens on box and Gaussian blur. I believe that the issue comes from the .frag shader files? I don't know.

For the time being, it use the _directional blur_ effects as it is miles more lightweight for some reason.

**Comment by itsmattkc** [comment]
As I saidin #205, saturating a GPU will fairly inevitably lead to freezing, there's not much that can be done about that. The blurs _can_ however still be heavily optimized.

I'd been aware of this some time after #205, that the blurs could be optimized by a two-step process. As @BrimsonBhin says, blurring one way then blurring another is much faster than trying to do both. The math is: averaging, say, 50 pixels horizontally and then 50 vertically is 100 operations, while averaging both at once is 50^2 = 2500 operations (and of course this gets dramatically worse the bigger the radius). I hadn't realized this while writing the blurs initially.

The two-step process is mathematically identical but requires infrastructure to run the shader twice. Luckily this was already mostly done, so in 8f9a3f37e3cadd636d2d628e1e4c255e6cb8a940 you should notice vastly improved blur performance using this two-step process.

The blurs could be optimized further, particularly Gaussian which uses a perfect Gaussian formula (fairly expensive, alternatives are using a lookup table, or probably more effectively using a cheaper function that approximates the Gaussian formula with only a small margin of error), but this should be a huge improvement.

**Comment by DaniSeeh** [comment]
Sorry, didn't mean to repeat the issue. Glad it can be sped up! Thank you.

**Comment by itsmattkc** [comment]
@DaniSeeh Haha don't worry, if you hadn't, this optimization would have probably happened a lot later.

**Comment by DaniSeeh** [comment]
Just tried the new build, what a difference! Night and day. And I love some of the new distorts you added.
